setup/src/setup-haskell.ts

```diff
@@ -4,6 +4,8 @@ import {getOpts, getDefaults, Tool} from './opts';
 import {installTool} from './installer';
 import type {OS} from './opts';
 import {exec} from '@actions/exec';
+import * as c from '@actions/cache';
+import {hashElement} from 'folder-hash';
```

Comment:
Do you think it would be better to use folder-hash or to copy in the exact code that the github runner uses for its [`hashFiles` function](https://github.com/actions/runner/blob/master/src/Misc/expressionFunc/hashFiles/src/hashFiles.ts)? I'm leaning towards the latter since it would (in theory) preserve hash compatibility if that code ever gets migrated/exposed in actions/toolkit.

Reply:
Right, that sounds more robust, I have copied `hashFiles`, but had to change it slightly to take normal arguments.